lightning/src/ln/channelmanager.rs

```diff
@@ -1203,6 +1204,37 @@ impl ChannelDetails {
 	}
 }
 
+/// Used to express the status of payments that are currently pending resolution. They do not
+/// include only payments that are currently inflight, but also those who may have yet to find a
+/// path or have outstanding HTLCs.
+#[derive(Debug, PartialEq)]
+pub enum RecentPaymentDetails {
+	/// When a payment is still being sent and awaiting successful delivery.
+	Pending {
+		/// Hash of the payment that is currently being sent but has yet to be fulfilled or
+		/// abandoned.
+		payment_hash: PaymentHash,
+		/// Total amount (in msat) across all paths for this payment, not just the amount currently
+		/// inflight.
+		total_msat: u64,
+	},
+	/// When a pending payment is fulfilled, we continue tracking it until all pending HTLCs have
+	/// been resolved. Upon receiving [`Event::PaymentSent`], we delay for a couple of seconds
+	/// before the payment is removed from tracking.
+	Fulfilled {
+		/// Hash of the payment that was claimed.
```

Comment:
Let's say when this would be `None`. Likely for older serializations.

Reply:
Fixed in 00fc748e